### ISSUE TITLE
Inconsistent parsing of data URLs in CSS in Next 14

### ISSUE THREAD

**Issue by prichey** [vercel/next.js#65853]
### Link to the code that reproduces this issue

[codemirror](https://codesandbox.io/p/devbox/css-bug-repro-4y4ktg?file=%2Fapp%2Fpage.tsx) / https://github.com/prichey/css-bug-repro

### To Reproduce

1. `next dev`
2. Note which divs have the incorrect background CSS (`.blue-bg`, `.pink-bg`).

### Current vs. Expected behavior

Using a 'partially' URL encoded SVG as data URL works as expected in `globals.css`, e.g.

```css
.red-bg {
  background-image: url("data:image/svg+xml,%3Csvg width='1' height='1' xmlns='http://www.w3.org/2000/svg'%3E%3Crect width='1' height='1' fill='red' /%3E%3C/svg%3E");
}
```

Using the same CSS in `styled-jsx` does not, e.g.

```jsx
<style jsx>{`
  .red-bg {
    background-image: url("data:image/svg+xml,%3Csvg width='1' height='1' xmlns='http://www.w3.org/2000/svg'%3E%3Crect width='1' height='1' fill='red' /%3E%3C/svg%3E");
  }
`}</style>
```

What's worse is that declarations following the 'broken' declaration are not respected, e.g.

```jsx
<style jsx>{`
  .red-bg {
    background-image: url("data:image/svg+xml,%3Csvg width='1' height='1' xmlns='http://www.w3.org/2000/svg'%3E%3Crect width='1' height='1' fill='red' /%3E%3C/svg%3E");
  }
  .pink-bg {
    background-color: pink; // not applied
  }
`}</style>
```

This worked without issue prior to updating to Next 14.2, but now is broken.

<img width="1512" alt="Screenshot 2024-05-16 at 11 12 43 AM" src="https://github.com/vercel/next.js/assets/1418487/de265b9a-c156-435b-8817-ce96ee16afb0">

As a sanity check, I downgraded to Next 13 (13.5.6), and it does indeed work as expected:

<img width="1512" alt="Screenshot 2024-05-16 at 11 26 36 AM" src="https://github.com/vercel/next.js/assets/1418487/5701a699-25e4-44ab-a380-81e9d28afd70">

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000
  Available memory (MB): 16384
  Available CPU cores: 8
Binaries:
  Node: 20.11.1
  npm: 10.2.4
  Yarn: 1.22.19
  pnpm: 9.1.0
Relevant Packages:
  next: 14.2.3 // Latest available version is detected (14.2.3).
  eslint-config-next: N/A
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure

### Which stage(s) are affected? (Select all that apply)

next dev (local), Other (Deployed)

**Comment by olaj** [comment]
I have the same issue.

It seems to work in [14.1.4](https://www.npmjs.com/package/next/v/14.1.4) but doesn't work in [14.2.0](https://www.npmjs.com/package/next/v/14.2.0)

**Comment by olaj** [comment]
@prichey Did you solve this? I still haven't updated because of this 🤷

**Comment by prichey** [comment]
@olaj Yeah, my solution was just to manually encode the SVG string. E.g.:

```tsx
<style jsx>{`
  .red-bg {
    background-image: url("data:image/svg+xml,${encodeURIComponent(
      `<svg width='1' height='1' xmlns='http://www.w3.org/2000/svg'><rect width='1' height='1' fill='red' /></svg>`
    )}");
  }
`}</style>
```

Hope that helps!

**Comment by marcoshernanz** [comment]
Thanks for the report. I reproduced this on 14.2.x, but it was fixed upstream in [swc-project/plugins#342](https://github.com/swc-project/plugins/pull/342) and reached Next.js in [#69605](https://github.com/vercel/next.js/pull/69605). The reproduction passes on Next.js 15 and 16, while Next.js 14 is now unsupported. Please upgrade to a supported release, so I’m closing this as fixed.
